op25/gr-op25_repeater/lib/p25p2_tdma.cc

```diff
@@ -606,23 +628,42 @@ int p25p2_tdma::handle_packet(uint8_t dibits[], const uint64_t fs)
 		xored_burst[i] = burstp[i] ^ tdma_xormask[sync.tdma_slotid() * BURST_SIZE + i];
 	}
 	if (burst_type == 0 || burst_type == 6)	{       // 4V or 2V burst
+		int current_slot = sync.tdma_slotid() >> 1;
+
 		track_vb(burst_type);
+
+		if (burst_type == 6)
+			d_tdma_slot_for_2v = current_slot;
+
+		if (current_slot <= (int) d_tdma_slot_for_2v)
+			current_slot += 5;
+
+		current_slot -= (d_tdma_slot_for_2v + 1);
+
+		if (current_slot != burst_id && current_slot > burst_id) {
+			int need_to_skip = current_slot - burst_id;
+			// XXX determine if the 2V frame was missed?
```

Comment:
As the comment says, can we use something else to determine other losses? Maybe the I-ISCH info...